__tests__/api/controllers/shoppingItem.controller.spec.ts

```diff
@@ -0,0 +1,171 @@
+  });
+  it('should find and fail when shoppingItem does not exist', async () => {
+    await request(express)
+      .get('/shoppingItems/100')
+      .expect(404);
+  });
+  it('should create shoppingItem', async () => {
+    doCreateItem();
+    const response = await request(express)
+      .post('/shoppingItems/')
+      .send(shoppingItemData)
+      .expect(201);
+    expect(response.body.price).toEqual(shoppingItemData.price);
+    expect(response.body.shop.id).toEqual(shoppingItemData.shopId);
+    expect(response.body.id).toBeGreaterThan(0);
+    expect(await ShoppingItem.findOne(response.body.id)).not.toBeUndefined();
+  });
+  it('should update primitive value of ShoppingItem', async () => {
+    doCreateItem();
+    let response = await request(express)
+      .post('/shoppingItems')
+      .send(shoppingItemData)
+      .expect(201);
+    expect(response.body.name).toEqual(shoppingItemData.name);
+    response = await request(express)
+      .get('/shoppingItems/' + response.body.id)
+      .expect(200);
+    const newPrice = 20.0;
+    shoppingItemData.price = newPrice;
+    response = await request(express)
+      .put('/shoppingItems/' + response.body.id)
+      .send(shoppingItemData)
+      .expect(200);
+    expect(response.body.price).toEqual(newPrice);
+  });
+  it('should update relation of ShoppingItem', async () => {
+    doCreateItem();
+    let response = await request(express)
+      .post('/shoppingItems')
+      .send(shoppingItemData)
+      .expect(201);
+    expect(response.body.name).toEqual(shoppingItemData.name);
+    response = await request(express)
+      .get('/shoppingItems/' + response.body.id)
+      .expect(200);
+    shoppingItemData.shopId = shoppingItemDataSetup.shopForUpdate.id;
+    response = await request(express)
+      .put('/shoppingItems/' + response.body.id)
+      .send(shoppingItemData)
+      .expect(200);
+    expect(response.body.shop.id).toEqual(shoppingItemData.shopId);
+  });
+  it('should delete an item', async () => {
+    doCreateItem();
+    let response = await request(express)
+      .post('/shoppingItems')
+      .send(shoppingItemData)
+      .expect(201);
+    response = await request(express)
+      .delete('/shoppingItems/' + response.body.id)
+      .expect(200);
+    expect(response.body.price).toEqual(shoppingItemData.price);
+    const itemAfterDeletion = await ShoppingItem.findOne(response.body.id);
+    expect(itemAfterDeletion).toBeUndefined();
+  });
+  it('should fail if update not existing item', async () => {
+    doCreateItem();
+    await request(express)
+      .put('/shoppingItems/100')
+      .send(shoppingItemData)
+      .expect(404);
+  });
+  it('should fail if delete not existing item', async () => {
+    await request(express)
+      .delete('/shoppingItems/100')
+      .expect(404);
+  });
+  it('should fail if store item without relation', async () => {
+    shoppingItemData.boughtForId = shoppingItemDataSetup.user.id;
```

Comment:
Here you can delete one relation from pack created in wrongly named method `doCreateItem` instead of assigning all of them but one.